source/php-frameworks/symfony.txt

```diff
@@ -0,0 +1,397 @@
+       Doctrine\Bundle\MongoDBBundle\DoctrineMongoDBBundle::class => ['all' => true],
+   ];
+
+Configure the ODM
+`````````````````
+
+In the ``config/packages`` directory, replace the contents of your
+``doctrine_mongodb.yaml`` file with the following code:
+
+.. code-block:: yaml
+   :caption: config/packages/doctrine_mongodb.yaml
+
+   doctrine_mongodb:
+       auto_generate_proxy_classes: true
+       auto_generate_hydrator_classes: true
+       connections:
+           default:
+               server: '%env(resolve:MONGODB_URL)%&appName=devrel.content.php'
+               options: {}
+       default_database: '%env(resolve:MONGODB_DB)%'
+       document_managers:
+           default:
+               auto_mapping: true
+               mappings:
+                   App:
+                       dir: '%kernel.project_dir%/src/Document'
+                       mapping: true
+                       type: attribute
+                       prefix: 'App\Document'
+                       is_bundle: false
+                       alias: App
+
+Install Frontend Dependency
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This project uses ``twig``, the default template engine for Symfony, to
+generate templates in this application. Run the following command to
+install the ``twig`` bundle:
+
+.. code-block::
+
+   composer require symfony/twig-bundle
+
+.. note::
+
+   This step might result in an error message about unset environment
+   variables, but this is resolved in the following section.
+
+Modify Project Files
+~~~~~~~~~~~~~~~~~~~~
+
+This section demonstrates how to modify the files in your
+``restaurants`` project to create a Symfony web application that views
+restaurants based on filter criteria.
+
+Set Environment Variables
+`````````````````````````
+
+In the root directory, navigate to the ``.env`` file and define the
+following environment variables or edit them if they are already present:
+
+.. code-block:: none
+   :caption: .env
+
+   ...
+
+   MONGODB_URL=<your Atlas connection string>
+   MONGODB_DB=sample_restaurants
+
+To retrieve your connection string, see the
+:ref:`php-symfony-atlas-cluster` step.
+
+Create Restaurant Entity and Controller
+```````````````````````````````````````
+
+Create the ``Restaurant.php`` file in the ``src/Document`` directory and
+paste the following code to create an entity that represents documents in
+the ``restaurants`` collection:
+
+.. literalinclude:: /includes/php-frameworks/Restaurant.php
+   :caption: src/Document/Restaurant.php
+   :language: php
+
+Next, create the ``RestaurantController.php`` file in the
+``src/Controller`` directory and paste the following code to handle the
+endpoints in your application:
+
+.. literalinclude:: /includes/php-frameworks/RestaurantController.php
+   :caption: src/Controller/RestaurantController.php
+   :language: php
+
+The controller file defines the ``index()`` method, which displays text on
+the web app's front page. The file also defines the ``browse()`` method,
```

Comment:
@GromNaN: Does it make more sense to refer to method names here instead of actions? I'll defer to you on this paragraph.